transaction/_manager.py

```diff
@@ -174,13 +175,15 @@ def run(self, func=None, tries=3):
         doc = func.__doc__
         if name != '_':
             if doc:
-                doc = name + '\n\n' + doc
+                doc = name + u'\n\n' + doc
             else:
                 doc = name
 
         for i in range(1, tries + 1):
             txn = self.begin()
             if doc:
+                if not isinstance(doc, text_type):
+                    doc = doc.decode('ascii')
```

Comment:
Do you know of a way to get the "coding" information? How do we know we're dealing with utf-8?